src/stdio_mgr/compat.py

```diff
@@ -0,0 +1,65 @@
+r"""``stdio_mgr.compat`` *code module*.
+
+``stdio_mgr.compat`` provides backports of Python standard library.
+
+**Author**
+    John Vandenberg (jayvdb@gmail.com)
+
+**File Created**
+    6 Sep 2019
+
+**Copyright**
+    \(c) Brian Skinn 2018-2019
+
+**Source Repository**
+    http://www.github.com/bskinn/stdio-mgr
+
+**Documentation**
+    See README.rst at the GitHub repository
+
+**License**
+    The Python-2.0 License.
+
+**Members**
+
+"""
+import abc
+
+# AbstractContextManager was introduced in Python 3.6
+# and may be used with typing.ContextManager.
+# See https://github.com/jazzband/contextlib2/pull/21 for more complete backport
+try:
+    from contextlib import AbstractContextManager
+except ImportError:  # pragma: no cover
```

Comment:
The error handling could be moved out to the caller, i.e.
```py
try:
    from contextlib import AbstractContextManager
except ImportError:
    from stdio_mgr.backports import AbstractContextManager
```

That has a slight advantage in that the code here can be unindented, allowing tests to be written and run on py36+ for the backport, but that also ends up equally complicated if there are several backports that are interdependent.